app/src/main/java/ch/epfl/sdp/map/GoogleMapApi.java

```diff
@@ -170,35 +147,41 @@ synchronized public void unDisplayEntity(Displayable displayable) {
         });
     }
 
-    private void removeMarkers(Displayable displayable) {
-        if (entityCircles.containsKey(displayable)) {
-            if (entityCircles.get(displayable).hasMarker())
-                entityCircles.get(displayable).getMarker().remove();
-            if (entityCircles.get(displayable).hasCircle())
-                entityCircles.get(displayable).getAoe().remove();
-        }
+    public void removeMarkers(Displayable displayable) {
+            if (entityCircles.containsKey(displayable)) {
+                if (entityCircles.get(displayable).hasMarker())
+                    entityCircles.get(displayable).getMarker().remove();
+                if (entityCircles.get(displayable).hasCircle())
+                    entityCircles.get(displayable).getAoe().remove();
+            }
     }
 
     public void displaySmallIcon(Displayable displayable, String title, int id) {
-        LatLng position = new LatLng(displayable.getLocation().getLatitude(), displayable.getLocation().getLongitude());
-        entityCircles.put(displayable, new MapDrawing(mMap.addMarker(new MarkerOptions()
-                .position(position)
-                .title(title)
-                .icon(BitmapDescriptorFactory.fromResource(id)))));
+        activity.runOnUiThread(() -> {

```

Comment:
I think this will need to be changed, as the advice told us the mapApi should be sure that everything is on the uiThread

Reply:
What do you mean?